internal/transport/http2_client.go

```diff
@@ -951,6 +954,9 @@ func (t *http2Client) Close(err error) {
 		t.mu.Unlock()
 		return
 	}
+	if logger.V(logLevel) {
+		logger.Infof("Closing transport, will close the connection. Err: %v", err)
```

Comment:
`transport: closing: %v` ?

Reply:
I think `will close the connection` is redundant since `Close` always closes the connection.

Reply:
loopyWriter exiting also always closes the connection, and you suggested a connection close explicit log message in that returning. Why not log that info here? Sorry can't seem to make this part of my atomic pr pass.

Reply:
I think the former is less obvious.  Closing the transport always closes the connection, as you would naturally expect, since the whole transport _should_ go away completely (including all running goroutines*) before returning, and the transport owns the connection.  Exiting the loopyWriter doesn't seem obviously tied to closing the connection, and I suspect if you asked the whole team 2 weeks ago, nobody would have known that off-hand (except I probably would have after thinking about it for a bit).

That's how I see it, anyway.  Let's make sure we have helpful logs in all the right places, but let's also make sure we don't over-clutter things with obvious information, either.

\* - I'm not sure if we do this correctly, but it _should_ happen

Reply:
Fine lol. Switched on both client and server side.